### PR TITLE
[STORY-10] Connect API to a real MongoDB database

### PR DESCRIPTION
**Pull request by biquando**
Story: #10

### CODE REVIEW

**Review by kevwjin — COMMENTED** [review]
@biquando should review these changes as they relate to user authentication.

File: api/routes/config.py [hunk 1]
```diff
@@ -15,12 +17,14 @@ async def export_config(
     server_name: str
```
Comment (sdcarlson) [inline api/routes/config.py:17]: would be a good idea to validate this input

File: api/routes/config.py [hunk 2]
```diff
@@ -32,15 +36,40 @@ async def import_config(
...
+            config_uid=server_uid
+        ))
+
+        db.user_collection.find_one_and_update(
+            {'username': current_user.username},
+            {'$set': {
+                'servers': [mapping.model_dump() for mapping in current_user.servers]
+            }}
+        )
     return 'Success'
```
Comment (sdcarlson) [inline api/routes/config.py:75]: return structured success response

File: api/routes/config.py (hunk 1 above)
Comment (biquando) [reply api/routes/config.py:17]: It is validated, the function goes through all of the user's server configs; if not config with the given name is found, then we return 404.

File: api/routes/config.py (hunk 2 above)
Comment (biquando) [reply api/routes/config.py]: Fixed

**Review by sdcarlson — APPROVED** [review]
LGTM

**Pull request merged**